packages/binding-file/src/file-client.ts

```diff
@@ -24,40 +24,48 @@ import path = require("path");
 
 const { debug, warn } = createLoggers("binding-file", "file-client");
 
+/**
+ * Used to determine the Content-Type of a file from the extension in its
+ * {@link filePath} if no explicit Content-Type is defined.
+ *
+ * @param filepath The file paththe Content-Type is determined for.
+ * @returns An appropriate Content-Type or `application/octet-stream` as a fallback.
+ */
+function mapFileExtensionToContentType(filepath: string) {
+    const fileExtension = path.extname(filepath);
+    debug(`FileClient found '${fileExtension}' extension`);
+    switch (fileExtension) {
+        case ".txt":
+        case ".log":
+        case ".ini":
+        case ".cfg":
+            return "text/plain";
+        case ".json":
+            return "application/json";
+        case ".jsontd":
+            return "application/td+json";
+        case ".jsonld":
+            return "application/ld+json";
+        default:
+            warn(`FileClient cannot determine media type for path '${filepath}'`);
+            return "application/octet-stream";
+    }
+}
+
 export default class FileClient implements ProtocolClient {
     public toString(): string {
         return "[FileClient]";
     }
 
+    private async readFile(filepath: string, contentType?: string): Promise<Content> {
+        const resource = fs.createReadStream(filepath);
+        const resourceContentType = contentType ?? mapFileExtensionToContentType(filepath);
+        return new Content(resourceContentType, resource);
+    }
+
     public async readResource(form: Form): Promise<Content> {
-        const filepath = form.href.split("//");
-        const resource = fs.createReadStream(filepath[1]);
-        const extension = path.extname(filepath[1]);
-        debug(`FileClient found '${extension}' extension`);
-        let contentType;
-        if (form.contentType != null) {
-            contentType = form.contentType;
-        } else {
-            // *guess* contentType based on file extension
-            contentType = "application/octet-stream";
-            switch (extension) {
-                case ".txt":
-                case ".log":
-                case ".ini":
-                case ".cfg":
-                    contentType = "text/plain";
-                    break;
-                case ".json":
-                    contentType = "application/json";
-                    break;
-                case ".jsonld":
-                    contentType = "application/ld+json";
-                    break;
-                default:
-                    warn(`FileClient cannot determine media type of '${form.href}'`);
-            }
-        }
-        return new Content(contentType, resource);
+        const filepath = form.href.split("//")[1];
```

Comment:
Would be failing if `href` split returns array with zero size, right? Are we fine with it?

Reply:
At the moment, this behavior is consistent with the previous version, I just refactored it a bit (previously, `filepath` was an array that got accessed this way twice). Adding a check or a fallback value sounds good to me, though – should this be done in this PR? Or should we follow up with another one to do so?

Reply:
I "think" we can also do it in this PR

Reply:
I now replaced the splitting in this line with accessing a parsed `URL` object. In this context, I noticed that the previous implementation might not have been completely correct since it used the *host* component of the parsed URL instead of the *path* component (which, at least according to [Wikipedia](https://en.wikipedia.org/wiki/File_URI_scheme) is "never correct"). It seems like the README of the file binding also needs to be adjusted in this regard.

Reply:
I remember that parsing File URIs is problematic. A couple of unit tests might be good. Can you add them?

Reply:
Sure, will do :)

Reply:
I now included the file client code in a separate PR: #1175 Once we have merged that one, I would perform a rebase to keep a clean and reviewable diff in this PR :)

Reply:
Makes sense 👍